Simplified/NYPLReportIssueViewController.swift

```diff
@@ -147,43 +147,33 @@ class NYPLReportIssueViewController: UITableViewController, UITextFieldDelegate,
   }
   
   override func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
-    
-    if HSAppearance.isIPad()
-    {
-      if(indexPath.row == 0){
-        return 44.0;
-      }else if(indexPath.row == 1){
-        return self.view.frame.size.height - 44.0;
-      }else{
-        return 44.0;
-      }
+
+    // height for the subject line of the message
+    let subjectHeight:CGFloat = 44.0
+
+    // height for the actual message body, it gets increased as we increase device screen size
+    var messageHeight:CGFloat = subjectHeight
+
+    let iPhonePadding:CGFloat  = 155.0
+    let iPadPadding:CGFloat    = 200.0
+
+    if indexPath.row == 0 {
+      return subjectHeight
     }
-    else{
-      if(indexPath.row == 0) {
-        return 44.0;
+    // we're at the message body
+    else if indexPath.row == 1 {
+      // is iPad
+      if HSAppearance.isIPad() {
```

Comment:
Can you think of how you would create different padding values based just on the current size of the table view, and not just whether it's an iPhone or iPad? With split screen, and larger iPhones like the Plus, a user is not guaranteed to see the correct layout.

[Apple has a good diagram as an example.](https://developer.apple.com/library/content/documentation/WindowsViews/Conceptual/AdoptingMultitaskingOniPad/QuickStartForSlideOverAndSplitView.html)

It would be something like this:

```
func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
    let subjectHeight:CGFloat = 44.0
    let heightOfTableView = tableView.frame.size.height

    if indexPath.row == 0 {
      return subjectHeight
    } else if indexPath.row == 1 {
      return heightOfTableView - subjectHeight * 2
    } else {
      return 0
    }
}
```